src/main/java/duke/ui/Parser.java

```diff
@@ -0,0 +1,155 @@
+            String[] parsed = Parser.parseDeadlineCommand(userInput);
+            c = new DeadlineCommand(parsed[0], parsed[1]);
+            break;
+        case EVENT:
+            parsed = Parser.parseEventCommand(userInput);
+            c = new EventCommand(parsed[0], parsed[1], parsed[2]);
+            break;
+        case FIND:
+            String toFind = Parser.parseFindCommand(userInput);
+            c = new FindCommand(toFind);
+            break;
+        default:
+        }
+        return c;
+    }
+
+    private static Commands getCommand(String userInput) throws DukeException {
+        try {
+            String strCommand = userInput.split(" ", 2)[0];
+            Commands command = Commands.valueOf(strCommand.toUpperCase());
+            return command;
+        } catch (IllegalArgumentException e) {
+            throw new DukeException("Invalid command");
+        } catch (NullPointerException e) {
+            throw new DukeException("Null exception encountered");
+        }
+    }
+
+    private static int parseMarkOrDeleteCommands(String userInput) throws DukeException {
+        String[] splitStr = userInput.split(" ", 2);
+        if (splitStr.length < 2) {
+            throw new DukeException("Mark / Unmark / Delete commands require an "
+                    + "integer argument referring to task number");
+        } else {
+            try {
+                int taskNumber = Integer.parseInt(splitStr[1]);
+                return taskNumber;
+            } catch (NumberFormatException e) {
+                throw new DukeException("Format of argument cannot be parsed as an integer");
+            }
+        }
+    }
+
+    private static String parseTodoCommand(String userInput) throws DukeException {
+        String[] splitStr = userInput.split(" ", 2);
+        if (splitStr.length < 2) {
+            throw new DukeException("Todo command requires a task description");
+        } else {
+            return splitStr[1];
+        }
+    }
+
+    private static String[] parseDeadlineCommand(String userInput) throws DukeException {
+        String[] splitStr = userInput.split(" ", 2);
+        if (splitStr.length < 2) {
+            throw new DukeException("duke.Task.Deadline command requires task description and /by argument");
+        } else {
+            String[] output = splitStr[1].split(" /by ", 2);
+            if (output.length < 2) {
+                throw new DukeException("duke.Task.Deadline command requires task description and /by argument");
+            } else {
+                return output;
+            }
+        }
+    }
+
+    private static String[] parseEventCommand(String userInput) throws DukeException {
+        String[] splitStr = userInput.split(" ", 2);
+        if (splitStr.length < 2) {
+            throw new DukeException("duke.Task.Event command requires task description, "
+                    + "/from argument and /to argument");
```

Comment:
I like how you have enough white spaces when splitting your code into two lines.